src/qibo/quantum_info/quantum_networks.py

```diff
@@ -42,7 +58,7 @@ def __init__(
         self.partition = partition
         self.system_output = system_output
         self._pure = pure
-        self._backend = backend
+        self._backend = backend if backend is not None else GlobalBackend()
```

Comment:
this line does not need to be changed because the backend check is already done inside `_set_tensor_and_parameters`